i3lock.c

```diff
@@ -277,6 +287,33 @@ static void input_done(void) {
     if (debug_mode)
         fprintf(stderr, "Authentication failure\n");
 
+    /* Get state of Caps and Num lock modifiers, to be displayed in
+     * STATE_PAM_WRONG state */
+    xkb_mod_index_t idx, num_mods;
+    const char *mod_name;
+    int used;
+
+    num_mods = xkb_keymap_num_mods(xkb_keymap);
+
+    for (idx = 0; idx < num_mods; idx++) {
+        if (!xkb_state_mod_index_is_active(xkb_state, idx, XKB_STATE_MODS_EFFECTIVE))
+            continue;
+
+        mod_name = xkb_keymap_mod_get_name(xkb_keymap, idx);
+        if (mod_name == NULL)
+            continue;
+        else if (strcmp(mod_name, XKB_MOD_NAME_CAPS) == 0)
```

Comment:
This is incomplete. There are other modifiers whose names are unclear to users, e.g. `XKB_MOD_NAME_LOGO` (= “Mod4”). Can you cover all the cases please?